### PR TITLE
SAMD21/SAMR21: Add more ADC lines

### PR DESCRIPTION
**Pull request by kbumsik**
This is a followup to #6929 .
This PR completes ADC line definitions of EXT headers for SAMD21/SAMR21 XPlained Pro.
I tested all lines using `test/periph_adc` and the conversion works well.

---
This change is [Reviewable](https://reviewable.io/reviews/riot-os/riot/7009)

### CODE REVIEW

File: boards/samr21-xpro/include/periph_conf.h [hunk 1]
```diff
@@ -240,6 +240,40 @@ static const spi_conf_t spi_config[] = {
...
+ */
+#define ADC_0_EN                           1
+#define ADC_MAX_CHANNELS                   14
+/* ADC 0 device configuration */
+#define ADC_0_DEV                          ADC
+#define ADC_0_IRQ                          ADC_IRQn
+/**
+ * @brief ADC Channel Configuration
+ */
+typedef struct {
```
Comment (aabadie) [inline boards/samr21-xpro/include/periph_conf.h]: This was taken from samd21-xpro but I think that this struct definition should be factorized `cpu/samd21/include/periph_cpu.h`.

Comment (haukepetersen) [inline boards/samr21-xpro/include/periph_conf.h]: +1

Comment (kbumsik) [inline boards/samr21-xpro/include/periph_conf.h]: You are absolutely right. I will move it.
Regarding type name, I think I need to change the type name to `adc_conf_chan_t` to match other peripheral struct names?

**Comment by aabadie** [comment]
Much better.
Can you split your commit in 3:
* one related to the move of adc_conf_chan_t to cpu with comment: "cpu/samd21: move adc channel struct definition to periph_cpu.h"
* one related to addition of channels for samd21-xpro: "boards/samd21-xpro: add more adc lines"
* one related to addition of ADC for samr21-xpro: "boards/samr21-xpro: add adc support"

Then it will be fine !

**Comment by kbumsik** [comment]
@aabadie Sure. Please check it.

**Comment by aabadie** [comment]
Looking at the datasheets, the changes are good for samr21-xpro and Murdock is happy, merging :)

**Pull request merged**